File: GitToolBox/src/test/java/zielu/gittoolbox/repo/GtConfigTest.java
package zielu.gittoolbox.repo;

import static org.junit.jupiter.api.Assertions.assertFalse;
import static org.junit.jupiter.api.Assertions.assertTrue;

import java.nio.file.Path;
import org.junit.jupiter.api.BeforeAll;
import org.junit.jupiter.api.Tag;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import zielu.TestType;
import zielu.junit5.intellij.extension.resources.ExternalPath;
import zielu.junit5.intellij.extension.resources.ResourcesExtension;

@Tag(TestType.FAST)
@ExtendWith(ResourcesExtension.class)
class GtConfigTest {
  private static GtConfig config;

  @BeforeAll
  static void beforeClass(@ExternalPath({".", "testData", "GtConfig", "config"}) Path testConfig) {
    config = GtConfig.load(testConfig.toFile());
  }

  @Test
  void isSvnRemote() throws Exception {
    assertTrue(config.isSvnRemote("svn"));
  }

  @Test
  void isRegularRemote() throws Exception {
    assertFalse(config.isSvnRemote("origin"));
  }
}

File: GitToolBox/src/test/kotlin/zielu/gittoolbox/repo/GtConfigTest.kt
package zielu.gittoolbox.repo

import org.junit.jupiter.api.BeforeAll
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.extension.ExtendWith
import zielu.junit5.intellij.extension.resources.ExternalPath
import zielu.junit5.intellij.extension.resources.ResourcesExtension
import java.nio.file.Path
import kotlin.test.assertFalse
import kotlin.test.assertTrue

@ExtendWith(ResourcesExtension::class)
internal class GtConfigTest {
  companion object {
    private lateinit var config: GtConfig

    @BeforeAll
    @JvmStatic
    fun beforeClass(@ExternalPath(".", "testData", "GtConfig", "config") testConfig: Path) {
      config = GtConfig.load(testConfig.toFile())
    }
  }

  @Test
  @Throws(Exception::class)
  fun isSvnRemote() {
    assertTrue { config.isSvnRemote("svn") }
  }

  @Test
  @Throws(Exception::class)
  fun isRegularRemote() {
    assertFalse { config.isSvnRemote("origin") }
  }
}
